core/lib/dal/src/transactions_dal.rs

```diff
@@ -818,7 +843,7 @@ impl TransactionsDal<'_, '_> {
             &l2_inputs as &[&[u8]],
             &l2_datas,
             &l2_tx_formats,
-            &l2_contract_addresses as &[&[u8]],
```

Comment:
Nit: do we need this change?

Reply:
I also updated the implementation for obtaining the `contract_address_bytes` to reduce the boilerplate.